packages/models/src/crypto-assets/crypto-asset.model.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  BtcCryptoAssetBalance,
+  CryptoAssetBalance,
+  StxCryptoAssetBalance,
+} from './crypto-asset-balance.model';
+import {
+  Brc20CryptoAssetInfo,
+  BtcCryptoAssetInfo,
+  InscriptionCryptoAssetInfo,
+  RuneCryptoAssetInfo,
+  Sip009CryptoAssetInfo,
+  Sip010CryptoAssetInfo,
+  StampCryptoAssetInfo,
+  Stx20CryptoAssetInfo,
+  StxCryptoAssetInfo,
+} from './crypto-asset-info.model';
+
+export type CryptoAssetType =
+  | 'btc'
+  | 'stx'
+  | 'brc-20'
+  | 'inscriptions'
+  | 'runes'
+  | 'sip-009'
+  | 'sip-010'
+  | 'src-20'
+  | 'stamps'
+  | 'stx-20';
+
+/**
+ * Cryptocurrencies - utility tokens
+ */
+export interface BtcCryptoAsset {
+  readonly blockchain: 'bitcoin';
+  readonly type: 'btc';
+  readonly asset: BtcCryptoAssetBalance & BtcCryptoAssetInfo;
+}
+
+export interface StxCryptoAsset {
+  readonly blockchain: 'stacks';
+  readonly type: 'stx';
+  readonly asset: StxCryptoAssetBalance & StxCryptoAssetInfo;
+}
+
+/**
+ * Fungible and non-fungible tokens
+ */
+export interface Brc20CryptoAsset {
+  readonly blockchain: 'bitcoin';
+  readonly type: 'brc-20';
+  readonly asset: CryptoAssetBalance & Brc20CryptoAssetInfo;
+}
+
+export interface RuneCryptoAsset {
+  readonly blockchain: 'bitcoin';
+  readonly type: 'runes';
+  readonly asset: CryptoAssetBalance & RuneCryptoAssetInfo;
```

Comment:
Similar to balance and market data, this pairs a single user's balance details together with the larger concept of the asset itself.

I find it helpful to think about building a database if Leather were a centralised app. We'd have one table for all the tokens supported. Another for user balances. The user balance table might reference the token for which it is a balance of. But you wouldn't put all token info and user balance data in the same table.

Putting these together, it's saying "in all possible cases, we'll pair a user's balance with the asset info". And that isn't necessarily the case. 

Consider in a signed out state we show all the tokens we support in the UI. We want to iterate though a list of known supported tokens, ideally as these types **but** oh this type expects us to have user balance, but there is no user. So, someone goes and creates dummy data like `const balance = createMoney(0, 'BTC') etc

Reply:
I don't think what you are describing here is accurate? 🤔 Yes, that is true if you use this model where they are coupled together, but they will also be available separately with their individual queries? I was thinking this would be used just to make using them together easier, but we can def just keep them separate. Is that your pref?

Reply:
I understand this now after a dm exchange, if coupling happens it should be done at the presentation layer not in our models.